apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/strategy/ApplyNovaTemplateOptionsCreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -129,8 +132,16 @@ protected ApplyNovaTemplateOptionsCreateNodesWithGroupEncodedIntoNameThenAddToSe
       ImmutableList.Builder<String> tagsBuilder = ImmutableList.builder();
 
       if (!templateOptions.getGroups().isEmpty()) {
+         Set<String> securityGroupNames = novaApi.getSecurityGroupApi(region).get().list()
+                 .transform(new Function<SecurityGroup, String>() {
+                    @Override
+                    public String apply(SecurityGroup input) {
+                       return input.getName();
+                    }
+                 })
+                 .toSet();
          for (String securityGroupName : templateOptions.getGroups()) {
-            checkNotNull(novaApi.getSecurityGroupApi(region).get().get(securityGroupName), "security group %s doesn't exist", securityGroupName);   
+            checkState(securityGroupNames.contains(securityGroupName), "Cannot find security group with name " + securityGroupName + ". \nSecurity groups available are: \n" + Iterables.toString(securityGroupNames)); // {
          }
```

Comment:
[minor] Change to `securityGroupNames.containsAll(templateOptions.getGroups())`.